LibTest/js_interop/JSAnyOperatorExtension/add_t06.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion JSAny add( JSAny? any )
+/// The result of `this + any` in JavaScript.
+///
+/// @description Check that `add` returns result of `this + any` in JavaScript.
+/// Test an array as `this`.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+import '../../../Utils/expect.dart';
+import '../js_utils.dart';
+
+test(JSAny? a) {
+  eval("var object = $a;");
+  eval(r'''
+                                  // If object == [1, 2] then:
+    var one = object + 1;         // String "1, 21"
+    var string = object + "s";    // String "1, 2s"
+    var a1 = object + [];         // String "1, 2"
+    var a2 = object + [1, 2];     // String "1, 21,2"
```

Comment:
Looks like JavaScript uses different `toString` semantics for the receiver and the argument. ;-)